File: product-ui/src/app/services/intelligence.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface IntelligenceEvaluateRequest {
  context?: string;
  inputs?: Record<string, unknown>;
}

export interface IntelligenceEvaluateResponse {
  outcome: string;
  confidence: number;
  payload?: unknown;
}

@Injectable({ providedIn: 'root' })
export class IntelligenceService {
  private readonly baseUrl = environment?.apiUrl ?? '/api';

  constructor(private readonly http: HttpClient) {}

  evaluate(request: IntelligenceEvaluateRequest): Observable<IntelligenceEvaluateResponse> {
    return this.http.post<IntelligenceEvaluateResponse>(
      `${this.baseUrl}/Intelligence/evaluate`,
      request
    );
  }
}

File: engine-api/src/EngineApi/Services/IntelligenceService.cs
using EngineApi.DTOs;

namespace EngineApi.Services;

public class IntelligenceService : IIntelligenceService
{
    public IntelligenceResponseDto Evaluate(IntelligenceRequestDto request)
    {
        if (request == null)
            throw new ArgumentNullException(nameof(request));
        return new IntelligenceResponseDto(
            Outcome: "evaluated",
            Confidence: 0.0,
            Payload: request.Inputs
        );
    }
}
